src/core/ir/mod.rs

```diff
@@ -54,6 +56,41 @@ impl IR {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Map {
+    pub input: Box<IR>,
+    // accept key return value instead of
+    pub map: BTreeMap<String, ConstValue>,
```

Comment:
```suggestion
    pub map: HashMap<String, String>,
```
Why need a const value?